lib/matplotlib/patches.py

```diff
@@ -806,6 +806,26 @@ def get_xy(self):
         """Return the left and bottom coords of the rectangle as a tuple."""
         return self._x0, self._y0
 
+    def get_edge_midpoints(self):
+        """
+        Return midpoints of the edges of the rectangle, moving anti-clockwise
+        from the center of the left-hand edge.
```

Comment:
`left-hand edge` doesn't play well with the rotation... for the corners, this is not an issue because, the "anchor point" (`xy`) is used, which is defines, but here this is not great.
I don't have any good suggestion, but maybe we can specify that this is before rotation... not very good but at least it has the benefit of being mentioned.

Reply:
Perhaps we should leave the midpoints method as private API given:
- I can't think of a good use case (apart from internally in widgets)
- It's tricky to explain

?

Reply:
Yes, private API is a good solution.